python/bindings/openravepy_robot.cpp

```diff
@@ -1878,11 +1886,12 @@ void init_openravepy_robot()
                         "jointinfos"_a,
                         "manipinfos"_a,
                         "attachedsensorinfos"_a,
+                        "connectedbodyinfos"_a,
                         "uri"_a = "",
                         DOXY_FN(RobotBase, Init)
                         )
 #else
-                      .def("Init", initrobot, Init_overloads(PY_ARGS("linkinfos", "jointinfos", "manipinfos", "attachedsensorinfos", "uri") DOXY_FN(RobotBase, Init)))
+                      .def("Init", initrobot, Init_overloads(PY_ARGS("linkinfos", "jointinfos", "manipinfos", "attachedsensorinfos", "connectedbodyinfos", "uri") DOXY_FN(RobotBase, Init)))
```

Comment:
@ziyan @bucketzxm Won't this break back compatibility by adding `connectedbodyinfos` in the middle? Do you have a set of merge requests to update the dependency codes on our side?

Reply:
We have some choices:

1. Move `connectedbodyinfos` to the end and make it optional
2. Make `JSONReader` a friend class of `RobotBase` and add connected bodies after `Init` is done
3. Maybe introduce a `RoboBasetInfo` and `KinBodyInfo` that has the other infos as members, I can see that there are additional things that might need to be added in the future.

Reply:
@ziyan i was thinking it's about time we make a proper Info structure (your #3 suggestion). How about you leave the current `Init` for back compat purposes and create a new `InitFromInfo` function that takes in a `RobotInfo` structure?

Reply:
Added `KinBodyInfo` and `RobotBaseInfo` and `InitFromInfo`.